src/client/java/teammates/client/scripts/sql/DataMigrationForUsageStatisticsSql.java

```diff
@@ -33,11 +45,43 @@ protected boolean isPreview() {
     }
 
     /**
-     * Always returns true, as the migration is needed for all entities from Datastore to CloudSQL .
+     * Queries for the latest SQL entity created, so that patching will only migrate newly created Datastore entities.
      */
     @Override
+    protected void setMigrationCriteria() {
+        if (!IS_PATCHING_MIGRATION) {
+            return;
+        }
+
+        HibernateUtil.beginTransaction();
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<Instant> cq = cb.createQuery(Instant.class);
+        Root<UsageStatistics> root = cq.from(UsageStatistics.class);
+        cq.select(cb.greatest(root.<Instant>get("createdAt")));
```

Comment:
Just change this to `startTime` since `createdAt` is what time the entity is created in the SQL DB. UsageStatistics are created one per hour, so this startTime should be unique as well